tests/tests_atmosphere/test_jacchia77.py

```diff
@@ -0,0 +1,289 @@
+        11.5638,
+        10.8698,
+        11.6540,
+        3.788,
+        -8.206,
+        -14.547,
+    ],
+    1500
+    * u.km: [
+        999.92,
+        0.6993,
+        -2.5839,
+        7.1525,
+        -8.7035,
+        10.8469,
+        10.6890,
+        11.0762,
+        2.776,
+        -8.783,
+        -15.260,
+    ],
+    2000
+    * u.km: [
+        999.97,
+        -3.7167,
+        -7.6281,
+        4.6303,
+        -9.9000,
+        10.2160,
+        10.5301,
+        10.7018,
+        1.986,
+        -9.158,
+        -15.780,
+    ],
+    2400
+    * u.km: [
+        999.98,
+        -6.8863,
+        -9.9000,
+        2.8201,
+        -9.9000,
+        9.7631,
+        10.4160,
+        10.5032,
+        1.553,
+        -9.357,
+        -16.148,
+    ],
+}
+
+
+@pytest.mark.parametrize("z", jacchia77_solutions.keys())
+def test_jacchia77(z):
+    #  Z, T, CN2, CO2, CO, CAr, CHe, CH, CM, WM)
+    expected_CN2 = jacchia77_solutions[z][1] * (u.m) ** -3
+    expected_CO2 = jacchia77_solutions[z][2] * (u.m) ** -3
+    expected_CO = jacchia77_solutions[z][3] * (u.m) ** -3
+    expected_CAr = jacchia77_solutions[z][4] * (u.m) ** -3
+    expected_CHe = jacchia77_solutions[z][5] * (u.m) ** -3
+    expected_CH = jacchia77_solutions[z][6] * (u.m) ** -3
+    expected_CM = jacchia77_solutions[z][7] * (u.m) ** -3
+    expected_WM = jacchia77_solutions[z][8] * (u.kg)
+
+    properties = Jacchia77().altitude_profile(z, 1000 * u.K)
+
+    for i in range(2, len(properties) - 1):
+        if properties[i].value > 1.26e-10:
+            properties[i] = np.log10(properties[i].value) * properties[i].unit
+        else:
+            properties[i] = -9.9 * properties[i].unit
+
+    Z, T, CN2, CO2, CO, CAr, CHe, CH, CM, WM = properties
+
+    assert_quantity_allclose(CN2, expected_CN2, rtol=1e-4)
+    assert_quantity_allclose(CO2, expected_CO2, rtol=1e-4)
+    assert_quantity_allclose(CO, expected_CO, rtol=1e-4)
+    assert_quantity_allclose(CAr, expected_CAr, rtol=1e-4)
+    assert_quantity_allclose(CHe, expected_CHe, rtol=1e-4)
+    assert_quantity_allclose(CH, expected_CH, rtol=1e-3)
+    assert_quantity_allclose(CM, expected_CM, rtol=1e-4)
+    assert_quantity_allclose(WM, expected_WM, rtol=1e-3)
+
+
+@pytest.mark.parametrize("z", jacchia77_solutions.keys())
+def test_tempertaure(z):
+    expected_T = jacchia77_solutions[z][0] * u.K
+    T = Jacchia77().temperature(z, 1000 * u.K)
+
+    assert_quantity_allclose(T, expected_T, rtol=1e-4)
+
+
+@pytest.mark.parametrize("z", jacchia77_solutions.keys())
+def test_pressure(z):
+    expected_p = jacchia77_solutions[z][9] * (u.N * u.m ** -2)
+    pressure = Jacchia77().pressure(z, 1000 * u.K)
+    p = np.log10(pressure.value) * pressure.unit
+
+    assert_quantity_allclose(p, expected_p, rtol=1e-3)
+
+
+@pytest.mark.parametrize("z", jacchia77_solutions.keys())
+def test_density(z):
+    expected_rho = jacchia77_solutions[z][10] * (
+        u.kg ** 2 * u.mol * u.K ** -1 * u.m ** -3
+    )
```

Comment:
@astrojuanlu Check this out ... I tried fixing this but got confused ... Will need to spend some time checking where the implementation has gone wrong as these are clearly not the units for `Density` 😞